app/models/manageiq/providers/kubernetes/container_manager/metrics_capture/capture_context.rb

```diff
@@ -0,0 +1,190 @@
+      @end_time = end_time
+      @interval = interval
+      @tenant = target.try(:container_project).try(:name) || '_system'
+      @ext_management_system = @target.ext_management_system
+      @ts_values = Hash.new { |h, k| h[k] = {} }
+
+      if @target.respond_to?(:hardware)
+        hardware = @target.hardware
+      else
+        hardware = @target.container_node.hardware
+      end
+
+      @node_cores = hardware.try(:cpu_total_cores)
+      @node_memory = hardware.try(:memory_mb)
+
+      validate_target
+    end
+
+    def collect_metrics
+      case @target
+      when ContainerNode then collect_node_metrics
+      when Container then collect_container_metrics
+      when ContainerGroup then collect_group_metrics
+      else raise TargetValidationError, "Validation error: unknown target"
+      end
+    end
+
+    private
+
+    CPU_NANOSECONDS = 1e09
+
+    def target_name
+      "#{@target.class.name.demodulize}(#{@target.id})"
+    end
+
+    def validate_target
+      unless @ext_management_system
+        raise TargetValidationError, "Validation error: ems not defined"
+      end
+      unless @node_cores.to_i > 0
+        raise TargetValidationError, "Validation error: cores not defined"
+      end
+      unless @node_memory.to_i > 0
+        raise TargetValidationError, "Validation error: memory not defined"
+      end
+    end
+
+    def collect_node_metrics
+      cpu_resid = "machine/#{@target.name}/cpu/usage"
+      process_cpu_counters_rate(fetch_counters_rate(cpu_resid))
+
+      mem_resid = "machine/#{@target.name}/memory/usage"
+      process_mem_gauges_data(fetch_gauges_data(mem_resid))
+    end
+
+    def collect_container_metrics
+      group_id = @target.container_group.ems_ref
+
+      cpu_resid = "#{@target.name}/#{group_id}/cpu/usage"
+      process_cpu_counters_rate(fetch_counters_rate(cpu_resid))
+
+      mem_resid = "#{@target.name}/#{group_id}/memory/usage"
+      process_mem_gauges_data(fetch_gauges_data(mem_resid))
+    end
+
+    def collect_group_metrics
+      group_id = @target.ems_ref
+
+      cpu_counters = @target.containers.collect do |c|
+        fetch_counters_rate("#{c.name}/#{group_id}/cpu/usage")
+      end
+      process_cpu_counters_rate(compute_summation(cpu_counters))
+
+      mem_gauges = @target.containers.collect do |c|
+        fetch_gauges_data("#{c.name}/#{group_id}/memory/usage")
+      end
+      process_mem_gauges_data(compute_summation(mem_gauges))
+    end
+
+    def hawkular_client
+      @client ||= Hawkular::Metrics::Client.new(
+        hakwular_entrypoint, hakwular_credentials, hakwular_options)
+    end
+
+    def hakwular_entrypoint
+      worker_class = ManageIQ::Providers::Kubernetes:: \
+                     ContainerManager::MetricsCollectorWorker
+      URI::HTTPS.build(
+        :host => @ext_management_system.hostname,
+        :port => worker_class.worker_settings[:hawkular_port],
+        :path => worker_class.worker_settings[:hawkular_path])
```

Comment:
yeah, @simon3z and I talked about the need for multi-endpoints here

agree this should probably be `metrics_`